homeassistant/components/adguard/switch.py

```diff
@@ -46,6 +69,75 @@ async def async_setup_entry(
     ]
     async_add_entities(switches, True)
 
+    async def async_service_handle(service_call: ServiceCall) -> None:
```

Comment:
My impression is that if you follow the steps in https://developers.home-assistant.io/docs/dev_101_services/#entity-services you don't need to implement any entity lookup logic at all, and this is entire handled by the service call platform (inside `homeassistant/helpers/service.py` i think.

I'd say remove this dispatcher method, and instead just add the individual methods on the entity and let the framework invoke them -- unless I am missing something happening here?